test_runner/fixtures/zenith_fixtures.py

```diff
@@ -1166,13 +1185,11 @@ def create(
         if not config_lines:
             config_lines = []
 
-        if branch is None:
-            branch = node_name
```

Comment:
This was the most unusual thing with the current approach, somewhat making the node (pg) <-> branch (pageserver) connection implicit.

So when we do 

```python
env.zenith_cli(["branch", "test_branch_segment_boundary", "test_branch_behind@0/3000000"])
pg = env.postgres.create_start("test_branch_segment_boundary")
```

`test_branch_segment_boundary` are different things, that are mapped together due to the assignment.

Reply:
Initially, I've wanted to get rid of node names entirely (and use branches instead), but after I had to return back to them and this is the most sane approach indeed, implemented the same way.